src/drivers/clock/sp804_timer/sp804_timer.c

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file sp804_timer.c
+ * @brief ARM Dual-Timer (SP804) Module driver
+ * @author Nishant Malpani <nish.malpani25@gmail.com>
+ * @date 2020-03-07
+ */
+
+#include <drivers/common/memory.h>
+#include <hal/clock.h>
+#include <hal/reg.h>
+#include <kernel/irq.h>
+#include <kernel/printk.h>
+#include <kernel/time/clock_source.h>
+#include <kernel/time/time_device.h>
+
+#include <framework/mod/options.h>
+#include <embox/unit.h>
+
+#define SP804_BASE      OPTION_GET(NUMBER, base_addr) /* Base address */
+#define SP804_IRQ       OPTION_GET(NUMBER, irq_nr) /* Interrupt Request number */
+#define SP804_TARGET_HZ OPTION_GET(NUMBER, target_freq) /* Frequency to run the timer at */
+#define SP804_TIMCLK    100000000 /* Frequency of the SP804 clock timer - 100 MHz */
+
+#define SP804_T1_LR     (SP804_BASE + 0x00) /* Timer1 Load Register */
+#define SP804_T1_VR     (SP804_BASE + 0x04) /* Timer1 Current Value Register */
+#define SP804_T1_CR     (SP804_BASE + 0x08) /* Timer1 Control Register */
+#define SP804_T1_ICR    (SP804_BASE + 0x0C) /* Timer1 Interrupt Clear Register */
+#define SP804_T1_RIS    (SP804_BASE + 0x10) /* Timer1 Raw Interrupt Status Register */
+#define SP804_T1_MIS    (SP804_BASE + 0x14) /* Timer1 Masked Interrupt Status Register */
+#define SP804_T1_BGL    (SP804_BASE + 0x18) /* Timer1 Background Load Register */
+
+#define SP804_CR_MASK           (0b11101111) /* Mask for bits 0-7 */
+#define SP804_CR_ONESHOT        (0 << 0) /* OneShot bit; 0 -> wrapping mode, 1 -> one-shot mode */
```

Comment:
What doe it mean?
Usually only (1 << bit) number using

Reply:
Here I mean to set the 0th bit with the value '0' - to choose the wrapping mode rather than the one-shot mode. I could have done it the way you suggested and done a 'NOT' when loading the appropriate register, although I thought this provides more readability to the driver's consumer (with the corresponding comments) while setting the timer parameters. 

I shall change if you think your way is more suitable.